lib/ansible/modules/net_tools/basics/uri.py

```diff
@@ -259,6 +259,41 @@
     method: POST
     src: /path/to/my/file.json
     remote_src: yes
+
+- name: Pause play until a URL is reachable from this host
+  uri:
+    url: "http://192.0.2.1/some/test"
+    follow_redirects: none
+    method: GET
+  register: _result
+  until: _result.status == 200
+  retries: 720 # 720 * 5 seconds = 1hour (60*60/5)
+  delay: 5 # Every 5 seconds
+
+# There are issues in a supporting Python library that is discussed in
+# https://github.com/ansible/ansible/issues/52705 where a proxy is defined
+# but you want to bypass proxy use on CIDR masks by using no_proxy
+- name: Work around a python issue that doesn't support no_proxy envvar
+  uri:
+    follow_redirects: none
+    validate_certs: False
+    timeout: 5
+    url: "http://{{ ip_address }}:{{ port | default(80) }}"
+  register: uri_data
+  failed_when: False
+  changed_when: False
+  vars:
+    ip_address: 192.0.2.1
+  environment: |
+      {
+        {% for no_proxy in (lookup('env', 'no_proxy') | replace(',', '') ).split() %}
+          {% if no_proxy| ipaddr | type_debug != 'NoneType' %}
```

Comment:
Shouldn't this better be:
```suggestion
          {% if no_proxy| ipaddr is not none %}
```
? (Similar for the next line.)

Reply:
Sorry, good point. That's a bad habit I have (it's littered through other code I've written!) I'll fix that one up now.